### DISCUSSION TITLE
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory in next js

### DISCUSSION THREAD

**Discussion by alim-2023** [vercel/next.js#50001] [App Router]
<--- Last few GCs --->

[6568:0000017EFCDA6C20]    59010 ms: Mark-Compact 1940.2 (2081.3) -> 1931.2 (2092.1) MB, 3454.4 / 0.0 ms  (average mu = 0.375, current mu = 0.197) allocation failure; scavenge might not succeed
[6568:0000017EFCDA6C20]    64201 ms: Mark-Compact 1944.6 (2095.8) -> 1939.9 (2100.1) MB, 4668.3 / 0.0 ms  (average mu = 0.239, current mu = 0.101) allocation failure; GC in old space requested

<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF7B79D075F node_api_throw_syntax_error+175087
 2: 00007FF7B7955366 v8::internal::wasm::WasmCode::safepoint_table_offset+66006
 3: 00007FF7B7956723 v8::internal::wasm::WasmCode::safepoint_table_offset+71059
 4: 00007FF7B848D511 v8::Isolate::ReportExternalAllocationLimitReached+65
 5: 00007FF7B8478166 v8::internal::V8::FatalProcessOutOfMemory+662
 6: 00007FF7B82DE880 v8::internal::EmbedderStackStateScope::ExplicitScopeForTesting+144
 7: 00007FF7B82DB89D v8::internal::Heap::CollectGarbage+4749
 8: 00007FF7B82F14C6 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath+2150
 9: 00007FF7B82F1DFF v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath+95
10: 00007FF7B8301020 v8::internal::Factory::NewFillerObject+448
11: 00007FF7B7FB69EF v8::internal::Runtime::SetObjectProperty+20719
12: 00007FF7B8536B61 v8::internal::SetupIsolateDelegate::SetupHeap+606705
13: 00007FF738B1BC6F

**Comment by icyJoseph** [comment]
Well, believe it or not, that's very little information. All we can derive from that is that, the process allocated so much memory, that it broke. 

What did you do? what is it your are using Next.js for? What commands did you run? Some info about your system would also be helpful.

**Reply by alim-2023** [reply]
We have used all of this 
   "axios": "^1.3.5",
    "bootstrap": "^5.2.3",
    "google-spreadsheet": "^3.3.0",
    "googleapis": "^118.0.0",
    "next": "13.2.4",
    "react": "^18.2.0",
    "react-bootstrap": "^2.7.2",
    "react-dom": "^18.2.0",
    "react-icons": "^4.8.0",
    "react-slick": "^0.29.0",
    "slick-carousel": "^1.8.1"

**Comment by nicholasgriffintn** [comment]
There's another issue for this here: https://github.com/vercel/next.js/issues/46756

It's definitely an issue and has been with RSC for some time now, waiting on a next response as to why these errors happen.

**Reply by alim-2023** [reply]
you have any solution for this .

**Reply by nicholasgriffintn** [reply]
Don't use app directory at the moment honestly, waiting on Vercel to actually resolve it.

I experience this error during pretty much every dev session against the app dir and have done since the initial beta. Doesn't appear to have improved at all over time.

**Reply by alim-2023** [reply]
I don't used app directory I have used page directory.
